pom.xml

```diff
@@ -1416,58 +1304,84 @@
-                        <staticGroups>*</staticGroups>
-                        <!-- To make static imports go on top: -->
-                        <staticAfter>false</staticAfter>
-                        <skip>${format.skip}</skip>
-                        <removeUnused>true</removeUnused>
+                        <upToDateChecking>
+                            <enabled>true</enabled>
+                            <indexFile>.cache/spotless-index-${version.spotless-maven-plugin}</indexFile>
+                        </upToDateChecking>
+                        <formats>
+                            <format>
+                                <!-- XML specific rules -->
+                                <includes>
+                                    <include>src/**/*.xml</include>
+                                </includes>
+                                <trimTrailingWhitespace/>
+                                <endWithNewline/>
+                                <licenseHeader>
+                                    <file>license.xml.header</file>
+                                    <!-- Delimiter is a regex which identifies the "top" of the file i.e. all above it is considered a header -->
+                                    <delimiter><![CDATA[(\n)*^(<[^!?]+|<!DOCTYPE.+)$]]></delimiter>
+                                    <skipLinesMatching><![CDATA[<\?xml.+\?>]]></skipLinesMatching>
+                                </licenseHeader>
+                                <indent>
+                                    <tabs>true</tabs>
+                                    <spacesPerTab>4</spacesPerTab>
+                                </indent>
+                            </format>
+                        </formats>
+                        <!-- define a language-specific format -->
+                        <java>
+                            <toggleOffOn/>
+                            <includes>
+                                <include>src/**/*.java</include>
+                            </includes>
+                            <excludes>
+                                <exclude>**/TypeHelper.java</exclude>
+                                <exclude>**/TypeHelperTest.java</exclude>
+                                <exclude>**/ConcurrentReferenceHashMap.java</exclude>
+                            </excludes>
+                            <indent>
+                                <tabs>true</tabs>
+                                <spacesPerTab>4</spacesPerTab>
+                            </indent>
+
+                            <importOrder>
+                                <!-- you can use an empty string for all the imports you didn't specify explicitly, '|' to join group without blank line, and '\#` prefix for static imports. -->
+                                <order>\#,java,javax,jakarta,org.openjdk.jmh,org.hibernate,org.hibernate.testing,org.hibernate.test,org.junit,org.jboss,com,javafx.beans,</order>
```

Comment:
Can this be imported from a file? People will need a file specifying import orders in order to set up formatting in their IDE.

Unless spotless gets executed automatically in the IDE, but I suspect it only is when compiling...

Reply:
yes there's an option to pass a file:
``` XML
<order>java|javax,org,com,com.diffplug,,\#com.diffplug,\#</order>  <!-- or use <file>${project.basedir}/eclipse.importorder</file> -->
```